tests/credential_plugins_test.py

```diff
@@ -162,3 +165,89 @@ def test_tss_import(self) -> None:
         ):
             # assert this module as opposed to older thycotic.secrets.server
             assert cls.__module__ == 'delinea.secrets.server'
+
+
+@pytest.mark.parametrize(
+    (
+        'url',
+        'app_id',
+        'object_query',
+        'object_query_format',
+        'reason',
+        'verify',
+    ),
+    [
+        (
+            'http://testurl.com',
+            'foobar123',
+            'foobar123',
+            'test',
+            'foobar123',
+            True,
+        ),
+        (
+            'http://testurl.com',
+            'foobar123',
+            'foobar123',
+            'test',
+            '',
+            True,
+        ),
+    ],
+)
+def test_aim_sensitive_traceback_masked(
+    mocker: MockerFixture,
+    monkeypatch: pytest.MonkeyPatch,
+    url: str,
+    app_id: str,
+    object_query: str,
+    object_query_format: str,
+    reason: str | None,
+    verify: bool,
+) -> None:
+    """Ensure that the sensitive information is not leaked in the traceback."""
+    my_response = requests.Response()
+    my_response.status_code = 404
+    my_response.url = 'not_found'
+
+    aim_request_mock = mocker.Mock(
+        autospec=True,
+        name='aim_request',
+        return_value=my_response,
+    )
+    monkeypatch.setattr(aim.requests, 'get', aim_request_mock)
+
+    if reason:
+        expected_url_in_exc = (
+            r'.*http://testurl\.com/AIMWebService/api/Accounts\?'
+            r'AppId=\*\*\*\*&Query=\*\*\*\*&QueryFormat=test&reason=\*\*\*\*.*'
+        )
+        expected_response_url_literal = (
+            'http://testurl.com/AIMWebService/api/Accounts?'
+            'AppId=****&Query=****&QueryFormat=test&reason=****'
+        )
+    else:
+        expected_url_in_exc = (
+            r'.*http://testurl\.com/AIMWebService/api/Accounts\?'
+            r'AppId=\*\*\*\*&Query=\*\*\*\*&QueryFormat=test.*'
+        )
+        expected_response_url_literal = (
+            'http://testurl.com/AIMWebService/api/Accounts?'
+            'AppId=****&Query=****&QueryFormat=test'
+        )
```

Comment:
It's best to avoid any logic in the tests. They should be as straightforward as possible. Doing otherwise complicates things unnecessarily and is prone to human error and accumulating complexity.

The solution is simple — parametrize these two vars just like `reason` so each parametrized test would have 3 inputs — reason and two expectations. You'd just pass them to the thing you're testing and check the assertion. No extra logic.